plugins/modules/fusion_ra.py

```diff
@@ -98,7 +98,16 @@
 )
 
 
-def human_to_principal(fusion, user_id):
+def get_principal(module, fusion):
```

Comment:
The original idea was that the user field was supposed to be replaced by a principal field, similar to the API and have it just accept users as well for convenience, which is a bit different than this implementation which treats it as a separate field which I think will be confusing to the SDK users. They should not collide as user name is a human readable string and generated principal IDs are UUID. But don't know if you guys did not make some other decision.

Reply:
We discussed this with @genalt and @LukasZahradnik and the result was that we should keep the `user` field for backward compatibility and align with our other SDKs by adding the `principal`. But I kinda agree with you that it might be more convenient for the user. @genalt and @LukasZahradnik WDYT?

Reply:
If it's for backward compatibility only, you can deprecate the old field so we can remove it in 2.0.0

Reply:
Also there is another story to support also `api-client` field to align with `hmctl role-assignment list --api-client`. It might become difficult to find out whether the input to principal field is user name or api-client name

Reply:
@jankodera-ps We decided to have both `user` and `principal` fields. It might be more confusing that the `principal` supports both arbitrary user name and generated principal ID. On top of that, you have to distinguish between those. Then there is also API Client name, which users might assume `principal` also supports..
@andreepyro We want to keep the `user` field.

Reply:
@jankodera-ps The idea was that all you need in the TF provider is the `principal` field, because users can provide the principal ID by using references (e.g., `fusion_api_client.my_client.id`)

Reply:
@LukasZahradnik sure, principal is actually in the api, user/api_client is convenience